api/v1alpha1/databasecluster_types.go

```diff
@@ -97,6 +97,8 @@ type Resources struct {
 type Engine struct {
 	// Type is the engine type
 	Type EngineType `json:"type"`
+	// UnsafeConfiguration field used to ensure bla bla bla
+	UnsafeConfiguration bool `json:"unsafeConfiguration,omitempty"`
```

Comment:
What do you think of putting this outside of the Engine? We might have "unsafe configs" for other components of the DBC CR.

Reply:
Also, I would name it `AllowUnsafeConfiguration` because setting this to true doesn't make the configuration inherently unsafe, you just allow for unsafe configs to be defined.

Reply:
Done. Thanks for catching this